### PR TITLE
[tensor] dist spec s2s uses all-to-all

### PR DESCRIPTION
**Pull request by ver217**
To optimize communication, dist spec s2s uses all-to-all.

### CODE REVIEW

File: colossalai/tensor/dist_spec_mgr.py [hunk 1]
```diff
@@ -68,6 +68,22 @@ def _gather(tensor: torch.Tensor, old_dist_spec: _DistSpec) -> torch.Tensor:
         assert len(buffer) == 1
         return buffer[0]
 
+    @staticmethod
+    def _all_to_all(tensor: torch.Tensor, old_dist_spec: _DistSpec, dist_spec: _DistSpec) -> torch.Tensor:
```
Comment (feifeibear) [inline colossalai/tensor/dist_spec_mgr.py:72]: Please add the necessary sanity check for the tensor device as well as the shape.

**Review by zxgx — APPROVED** [review]

File: colossalai/tensor/dist_spec_mgr.py (hunk 1 above)
Comment (ver217) [reply colossalai/tensor/dist_spec_mgr.py:72]: done

**Review by zxgx — APPROVED** [review]

**Review by feifeibear — APPROVED** [review]

**Comment by feifeibear** [comment]
test_tensor目录下单测哦度过了
image

File: colossalai/tensor/dist_spec_mgr.py [hunk 2]
```diff
@@ -43,8 +43,25 @@ class DistSpecManager:
 
     _use_autograd_function: bool = True
 
+    def _sanity_check(old_dist_spec: _DistSpec, dist_spec: _DistSpec) -> None:
```
[the remark this answers is not in the document]
Comment (zxgx) [reply colossalai/tensor/dist_spec_mgr.py:46]: after `@staticmethod` added
image

**Pull request merged**